bin/reth/src/node/mod.rs

```diff
@@ -600,392 +236,11 @@ impl<Ext: RethCliExt> NodeCommand<Ext> {
-                    TotalDifficultyStage::new(consensus)
-                        .with_commit_threshold(config.total_difficulty.commit_threshold),
-                )
-                .set(SenderRecoveryStage {
-                    commit_threshold: config.sender_recovery.commit_threshold,
-                })
-                .set(
-                    ExecutionStage::new(
-                        factory,
-                        ExecutionStageThresholds {
-                            max_blocks: config.execution.max_blocks,
-                            max_changes: config.execution.max_changes,
-                            max_cumulative_gas: config.execution.max_cumulative_gas,
-                        },
-                        config
-                            .merkle
-                            .clean_threshold
-                            .max(config.account_hashing.clean_threshold)
-                            .max(config.storage_hashing.clean_threshold),
-                        prune_modes.clone(),
-                    )
-                    .with_metrics_tx(metrics_tx),
-                )
-                .set(AccountHashingStage::new(
-                    config.account_hashing.clean_threshold,
-                    config.account_hashing.commit_threshold,
-                ))
-                .set(StorageHashingStage::new(
-                    config.storage_hashing.clean_threshold,
-                    config.storage_hashing.commit_threshold,
-                ))
-                .set(MerkleStage::new_execution(config.merkle.clean_threshold))
-                .set(TransactionLookupStage::new(
-                    config.transaction_lookup.commit_threshold,
-                    prune_modes.transaction_lookup,
-                ))
-                .set(IndexAccountHistoryStage::new(
-                    config.index_account_history.commit_threshold,
-                    prune_modes.account_history,
-                ))
-                .set(IndexStorageHistoryStage::new(
-                    config.index_storage_history.commit_threshold,
-                    prune_modes.storage_history,
-                )),
-            )
-            .build(provider_factory);
-
-        Ok(pipeline)
-    }
-
-    /// Change rpc port numbers based on the instance number.
-    fn adjust_instance_ports(&mut self) {
-        // auth port is scaled by a factor of instance * 100
-        self.rpc.auth_port += self.instance * 100 - 100;
-        // http port is scaled by a factor of -instance
-        self.rpc.http_port -= self.instance - 1;
-        // ws port is scaled by a factor of instance * 2
-        self.rpc.ws_port += self.instance * 2 - 2;
-    }
 }
 
 /// Drives the [NetworkManager] future until a [Shutdown](reth_tasks::shutdown::Shutdown) signal is
 /// received. If configured, this writes known peers to `persistent_peers_file` afterwards.
-async fn run_network_until_shutdown<C>(
+pub(crate) async fn run_network_until_shutdown<C>(
```

Comment:
we should move this somewhere else

Reply:
added a `run_until_graceful_shutdown` method which takes a `FnOnce` to run. Writing the peers to a file is extracted into its own method, which is now in `bin/reth/src/utils.rs`.